### PR TITLE
take advantage of upcoming major version release to remove deprecated things

### PR DESCRIPTION
**Pull request by vandonr-amz**
previous major release was [7.0.0](https://github.com/apache/airflow/releases/tag/providers-amazon%2F7.0.0), and was cut on Jan 5 2023

So I removed all the obsolete warnings that were added in 2022 or before, ensuring users have had at least 1 full major version to migrate.

### CODE REVIEW

**Review by vincbeck — APPROVED** [review]
<3

File: docs/apache-airflow-providers-amazon/operators/redshift/redshift_sql.rst [hunk 1]
```diff
@@ -1,64 +0,0 @@
...
-``RedshiftSQLOperator`` executes a SQL query against an Amazon Redshift cluster using a Postgres connection.
-
-To execute a SQL query against an Amazon Redshift cluster without using a Postgres connection,
-please check ``RedshiftDataOperator``.
-
-.. exampleinclude:: /../../tests/system/providers/amazon/aws/example_redshift.py
-    :language: python
-    :dedent: 4
-    :start-after: [START howto_operator_redshift_sql]
-    :end-before: [END howto_operator_redshift_sql]
```
Comment (eladkal) [inline docs/apache-airflow-providers-amazon/operators/redshift/redshift_sql.rst:49]: If we remove the doc we need also to remove the example dag.

File: docs/apache-airflow-providers-amazon/operators/redshift/redshift_sql.rst [hunk 2]
```diff
@@ -1,64 +0,0 @@
...
-.. _howto/operator:RedshiftSQLOperator:
-
-===================
-Amazon Redshift SQL
-===================
-
-`Amazon Redshift <https://aws.amazon.com/redshift/>`__ manages all the work of setting up, operating, and scaling a data warehouse:
-provisioning capacity, monitoring and backing up the cluster, and applying patches and upgrades to
-the Amazon Redshift engine. You can focus on using your data to acquire new insights for your
-business and customers.
```
Comment (eladkal) [inline docs/apache-airflow-providers-amazon/operators/redshift/redshift_sql.rst]: I suggest to think if we want to remove this doc.
Maybe we should modify it to show how to use Redshift with `SQLExecuteQueryOperator` ?
If we remove the doc then users who land in Amazon docs may not be aware of the option to query redshift via `SQLExecuteQueryOperator`

**Comment by eladkal** [comment]
Also please add a note to the changelog with some information about what was removed and what users should do to mitigate. You can see examples in [other providers ](https://github.com/apache/airflow/blob/fbc13825140cc6cc6b3d4b27db6d329692a1c451/airflow/providers/google/CHANGELOG.rst#L47) This is important so that users will have easier life when upgrading

File: docs/apache-airflow-providers-amazon/operators/redshift/redshift_sql.rst (hunk 1 above)
Comment (vandonr-amz) [reply docs/apache-airflow-providers-amazon/operators/redshift/redshift_sql.rst:49]: oh yes right

File: docs/apache-airflow-providers-amazon/operators/redshift/redshift_sql.rst (hunk 2 above)
Comment (vandonr-amz) [reply docs/apache-airflow-providers-amazon/operators/redshift/redshift_sql.rst]: ok, can I write this in a subsequent PR ?

Comment (eladkal) [reply docs/apache-airflow-providers-amazon/operators/redshift/redshift_sql.rst]: Preferably not.. I think this one is important.
What we can do is simply not remove the operator in this cycle thus leaving it with the docs as is. This is the fastest action

**Review by eladkal — APPROVED** [review]

**Review by potiuk — APPROVED** [review]

**Pull request merged**